cli/tools/lint.rs

```diff
@@ -89,27 +104,75 @@ pub async fn lint_files(
     rules_exclude,
   )?;
 
-  let has_error = Arc::new(AtomicBool::new(false));
-
-  let reporter_kind = if json {
-    LintReporterKind::Json
-  } else {
-    LintReporterKind::Pretty
+  let resolver = |changed: Option<Vec<PathBuf>>| {
+    let files_changed = changed.is_some();
+    let result =
+      collect_files(&args.clone(), &ignore.clone(), is_supported_ext).map(
+        |files| {
+          if let Some(paths) = changed {
+            files
+              .into_iter()
+              .filter(|path| paths.contains(path))
+              .collect::<Vec<_>>()
+          } else {
+            files
+          }
+        },
+      );
+    let paths_to_watch = args.clone();
+    async move {
+      if (files_changed || !watch)
+        && matches!(result, Ok(ref files) if files.is_empty())
+      {
+        ResolutionResult::Ignore
+      } else {
+        ResolutionResult::Restart {
+          paths_to_watch,
+          result,
+        }
+      }
+    }
   };
-  let reporter_lock = Arc::new(Mutex::new(create_reporter(reporter_kind)));
 
-  let no_of_files_linted =
+  let operation = |paths: Vec<PathBuf>| async {
+    let target_files_len = paths.len();
+    let reporter_kind = reporter_kind.clone();
+    let reporter_lock = Arc::new(Mutex::new(create_reporter(reporter_kind)));
+    run_parallelized(paths, {
+      let has_error = has_error.clone();
+      let lint_rules = lint_rules.clone();
+      let reporter_lock = reporter_lock.clone();
+      move |file_path| {
+        let r = lint_file(file_path.clone(), lint_rules.clone());
+        handle_lint_result(
+          &file_path.to_string_lossy(),
+          r,
+          reporter_lock.clone(),
+          has_error,
+        );
+
+        Ok(())
+      }
+    })
+    .await?;
+    reporter_lock.lock().unwrap().close(target_files_len);
+
+    Ok(())
+  };
+  if watch {
+    file_watcher::watch_func(resolver, operation, "Lint").await?;
+  } else {
     if args.len() == 1 && args[0].to_string_lossy() == "-" {
+      let reporter_lock =
+        Arc::new(Mutex::new(create_reporter(reporter_kind.clone())));
       let r = lint_stdin(lint_rules);
```

Comment:
I believe there might be an edge case when running `deno lint --watch -`, this will error out

Reply:
![image](https://user-images.githubusercontent.com/15936231/134757161-84271136-700d-4ead-b603-8499da8a2023.png)

I tried without any error, should I add a guard to tell the user cannot watch stdin?

Reply:
> I tried without any error, should I add a guard to tell the user cannot watch stdin?

I think that makes sense, please do.